rosbag2_cpp/include/rosbag2_cpp/writers/sequential_writer.hpp

```diff
@@ -118,6 +125,8 @@ class ROSBAG2_CPP_PUBLIC SequentialWriter
   std::shared_ptr<rosbag2_cpp::cache::MessageCache> message_cache_;
   std::unique_ptr<rosbag2_cpp::cache::CacheConsumer> cache_consumer_;
 
+  std::shared_ptr<rosbag2_cpp::cache::CircularMessageCache> circular_message_cache_;
```

Comment:
I don't think it makes sense to have two separate caches here - it makes a lot more sense to have a single `MessageCache` interface that is either polymorphic or templated to do what we need.